mods/ra/maps/allies-08a/allies08a.lua

```diff
@@ -0,0 +1,148 @@
+	end)
+end
+
+ticked = TimerTicks
+Tick = function()
+	ussr.Cash = 5000
+
+	if ussr.HasNoRequiredUnits() then
+		greece.MarkCompletedObjective(DefendChronosphere)
+		greece.MarkCompletedObjective(KeepBasePowered)
+	end
+
+	if greece.HasNoRequiredUnits() then
+		ussr.MarkCompletedObjective(BeatAllies)
+	end
+
+	if ticked > 0 then
+		UserInterface.SetMissionText("Chronosphere experiment completes in " .. Utils.FormatTime(ticked), TimerColor)
+		ticked = ticked - 1
+	elseif ticked == 0 and (greece.PowerState ~= "Normal") then
+		greece.MarkFailedObjective(KeepBasePowered)
+	elseif ticked == 0 then
+		DefendChronosphereCompleted()
+		ticked = ticked - 1
+	end
+end
+
+WorldLoaded = function()
+	greece = Player.GetPlayer("Greece")
+	ussr = Player.GetPlayer("USSR")
+	germany = Player.GetPlayer("Germany")
+
+	DefendChronosphere = greece.AddPrimaryObjective("Defend the Chronosphere and the Tech Center \nat all costs.")
+	KeepBasePowered = greece.AddPrimaryObjective("The Chronosphere must have power when the \ntimer runs out.")
+	EvacuateScientists = greece.AddSecondaryObjective("Evacuate scientists from the island to \nthe Tech Center.")
+	BeatAllies = ussr.AddPrimaryObjective("Defeat the Allied forces.")
+
+	Trigger.OnObjectiveCompleted(greece, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(greece, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerLost(greece, function()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			Media.PlaySpeechNotification(greece, "MissionFailed")
+		end)
+	end)
+	Trigger.OnPlayerWon(greece, function()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			Media.PlaySpeechNotification(greece, "MissionAccomplished")
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(1), function()
+		Media.PlaySpeechNotification(greece, "TwentyMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(11), function()
+		Media.PlaySpeechNotification(greece, "TenMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(16), function()
+		Media.PlaySpeechNotification(greece, "WarningFiveMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(18), function()
+		Media.PlaySpeechNotification(greece, "WarningThreeMinutesRemaining")
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(20), function()
+		Media.PlaySpeechNotification(greece, "WarningOneMinuteRemaining")
+	end)
+
+	PowerProxy = Actor.Create("powerproxy.paratroopers", false, { Owner = ussr })
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+	TimerColor = greece.Color
+
+	Trigger.OnAnyKilled(ObjectiveBuildings, function()
+		greece.MarkFailedObjective(DefendChronosphere)
+	end)
+
+	Trigger.OnEnteredFootprint(ScientistFootprint, function(a, id)
+		if a.Owner == greece and not scientistsTriggered then
+			scientistsTriggered = true
+			Trigger.RemoveFootprintTrigger(id)
+			CreateScientists()
+		end
+	end)
```

Comment:
Attempted to add the scientist rescue, but I'm getting a crash on entering the footprint. Best I can tell it didn't like what I did here.